Sprint 7

RealMadSquirrel opened the pull request:
Финальный проект 7 спринта

Commit:
Финальный проект спринта №7

    -- Тесты на ручку «Создать курьера».
    -- Тесты на ручку «Логин курьера».
    -- Тесты на ручку «Создать заказ».
    -- Тесты на ручку, которая получает список заказов.

Commit:
Финальный проект спринта №7

    -- Добавлен allure-отчет

Commit:
Финальный проект спринта №7

    -- Добавлен новый allure-отчет
    -- Изменение в названии тестов в файле test_order.py(была ошибка)

AlexKitIt commented on conftest.py:
```diff
@@ -0,0 +1,8 @@
+import pytest
+import helper
+
+''''@pytest.fixture(scope='function')
+def create_courier():
+    courier_response = helper.register_new_courier_and_return_login_password()
+
+    return courier_response'''
```
Нужно исправить: почему фикстура в комментах?

AlexKitIt commented on tests/test_create_courier.py:
```diff
@@ -0,0 +1,27 @@
+        creds = CourierFactory.register_new_courier_and_return_login_password()
+        CourierApi.create_courier(creds)
+        created_courier_request_2 = CourierApi.create_courier(creds)
+        assert (created_courier_request_2.status_code == 409 and created_courier_request_2.json()["message"] == 'Этот логин уже используется. Попробуйте другой.')
```
Можно лучше: тексты ответов стоит хранить снаружи в data модулях для поддержания и актуализации

AlexKitIt commented on tests/test_login_courier.py:
```diff
@@ -0,0 +1,33 @@
+    def test_success_login_courier(self):
+        creds = CourierFactory.register_new_courier_and_return_login_password()
+        created_courier_request = CourierApi.create_courier(creds)
+        if created_courier_request.status_code == 201:
```
Нужно исправить: условий в тестах быть не может, тест всегда однозначен

AlexKitIt commented on tests/test_login_courier.py:
```diff
@@ -0,0 +1,33 @@
+    @allure.title("Проверка успешной авторизации")
+    @allure.description("Создание шаблонного курьера, проверка статуса ответа. Авторизация, удаление.")
+    def test_success_login_courier(self):
+        creds = CourierFactory.register_new_courier_and_return_login_password()
```
Можно лучше: в тестах авторизации регистрация будет предусловием, а они у нас живут в conftest

Commit:
Доработка №1 финального проект спринта №7

    -- Добавлен новый allure-отчет версия 7_2
    -- Добавлена фикстура (данные для создания курьера)
    -- добавлена фикстура в тесты по созданию курьера и логированию
    -- вынесены тексты ответов в data
